.github/workflows/app_build.yml

```diff
@@ -253,32 +274,39 @@ jobs:
           echo MACOSX_DEPLOYMENT_TARGET=${{ matrix.MACOSX_DEPLOYMENT_TARGET }} >> $GITHUB_ENV
           echo SDKROOT=${{ matrix.SDKROOT }} >> $GITHUB_ENV
           echo CMAKE_MACOSX_DEPLOYMENT_TARGET='-DCMAKE_OSX_DEPLOYMENT_TARGET=$MACOSX_DEPLOYMENT_TARGET' >> $GITHUB_ENV
-           # The MACOSX_DEPLOYMENT_TARGET environment variable sets the default value for the CMAKE_OSX_DEPLOYMENT_TARGET variable.
+          # The MACOSX_DEPLOYMENT_TARGET environment variable sets the default value for the CMAKE_OSX_DEPLOYMENT_TARGET variable.
           # echo CMAKE_MACOSX_DEPLOYMENT_TARGET='-DCMAKE_OSX_DEPLOYMENT_TARGET=$MACOSX_DEPLOYMENT_TARGET' >> $GITHUB_ENV
-          echo "Using brew to install ninja"
-          brew install ninja
-          echo CMAKE_GENERATOR='Ninja' >> $GITHUB_ENV
 
-          brew install md5sha1sum
-
-          curl -L -O https://download.qt.io/official_releases/qt-installer-framework/4.3.0/QtInstallerFramework-macOS-x64-4.3.0.dmg
-          hdiutil attach -mountpoint ./qtfiw_installer QtInstallerFramework-macOS-x64-4.3.0.dmg
-          echo "ls ./qtfiw_installer"
-          ls ./qtfiw_installer
-          echo "ls ./qtfiw_installer/QtInstallerFramework-macOS-x64-4.3.0.app/"
-          ls ./qtfiw_installer/QtInstallerFramework-macOS-x64-4.3.0.app/
-          echo "ls ./qtfiw_installer/QtInstallerFramework-macOS-x64-4.3.0.app/Contents/"
-          ls ./qtfiw_installer/QtInstallerFramework-macOS-x64-4.3.0.app/Contents/
-          echo "ls ./qtfiw_installer/QtInstallerFramework-macOS-x64-4.3.0.app/Contents/MacOS"
-          ls ./qtfiw_installer/QtInstallerFramework-macOS-x64-4.3.0.app/Contents/MacOS
-          echo "ls ./qtfiw_installer/QtInstallerFramework-macOS-x64-4.3.0.app/Contents/Resources"
-          ls ./qtfiw_installer/QtInstallerFramework-macOS-x64-4.3.0.app/Contents/Resources
-
-          sudo ./qtfiw_installer/QtInstallerFramework-macOS-x64-4.3.0.app/Contents/MacOS/QtInstallerFramework-macOS-x64-4.3.0 --verbose --script ./ci/install_script_qtifw.qs
-          ls ~
-          ls ~/Qt/ || true
-          ls ~/Qt/QtIFW-4.3.0 || true
-          echo "~/Qt/QtIFW-4.3.0/bin/" >> $GITHUB_PATH
+          if [ "${{matrix.SELF_HOSTED}}" = "true" ]; then
+            echo "Using previously installed ninja and IFW"
+            echo "/Users/irvinemac/Qt/Tools/QtInstallerFramework/4.3/bin/" >> $GITHUB_PATH
+          else
```

Comment:
(maybe we should have a dedicated workflow yml file for the self hosted?)

Reply:
I thought about both of those.  I wanted to try the caching and see how it goes, seems like it's not worth it.  I'd prefer to keep a single workflow file otherwise I'm sure they'll diverge.